samplers/jaegerremote/internal/proto-gen/jaeger-idl/proto/api_v2/sampling.pb.go

```diff
@@ -0,0 +1,1568 @@
+	if deterministic {
+		return xxx_messageInfo_SamplingStrategyResponse.Marshal(b, m, deterministic)
+	} else {
+		b = b[:cap(b)]
+		n, err := m.MarshalToSizedBuffer(b)
+		if err != nil {
+			return nil, err
+		}
+		return b[:n], nil
+	}
+}
+func (m *SamplingStrategyResponse) XXX_Merge(src proto.Message) {
+	xxx_messageInfo_SamplingStrategyResponse.Merge(m, src)
+}
+func (m *SamplingStrategyResponse) XXX_Size() int {
+	return m.Size()
+}
+func (m *SamplingStrategyResponse) XXX_DiscardUnknown() {
+	xxx_messageInfo_SamplingStrategyResponse.DiscardUnknown(m)
+}
+
+var xxx_messageInfo_SamplingStrategyResponse proto.InternalMessageInfo
+
+func (m *SamplingStrategyResponse) GetStrategyType() SamplingStrategyType {
+	if m != nil {
+		return m.StrategyType
+	}
+	return SamplingStrategyType_PROBABILISTIC
+}
+
+func (m *SamplingStrategyResponse) GetProbabilisticSampling() *ProbabilisticSamplingStrategy {
+	if m != nil {
+		return m.ProbabilisticSampling
+	}
+	return nil
+}
+
+func (m *SamplingStrategyResponse) GetRateLimitingSampling() *RateLimitingSamplingStrategy {
+	if m != nil {
+		return m.RateLimitingSampling
+	}
+	return nil
+}
+
+func (m *SamplingStrategyResponse) GetOperationSampling() *PerOperationSamplingStrategies {
+	if m != nil {
+		return m.OperationSampling
+	}
+	return nil
+}
+
+type SamplingStrategyParameters struct {
+	ServiceName          string   `protobuf:"bytes,1,opt,name=serviceName,proto3" json:"serviceName,omitempty"`
+	XXX_NoUnkeyedLiteral struct{} `json:"-"`
+	XXX_unrecognized     []byte   `json:"-"`
+	XXX_sizecache        int32    `json:"-"`
+}
+
+func (m *SamplingStrategyParameters) Reset()         { *m = SamplingStrategyParameters{} }
+func (m *SamplingStrategyParameters) String() string { return proto.CompactTextString(m) }
+func (*SamplingStrategyParameters) ProtoMessage()    {}
+func (*SamplingStrategyParameters) Descriptor() ([]byte, []int) {
+	return fileDescriptor_ae32d90db01957f7, []int{5}
+}
+func (m *SamplingStrategyParameters) XXX_Unmarshal(b []byte) error {
+	return m.Unmarshal(b)
+}
+func (m *SamplingStrategyParameters) XXX_Marshal(b []byte, deterministic bool) ([]byte, error) {
+	if deterministic {
+		return xxx_messageInfo_SamplingStrategyParameters.Marshal(b, m, deterministic)
+	} else {
+		b = b[:cap(b)]
+		n, err := m.MarshalToSizedBuffer(b)
+		if err != nil {
+			return nil, err
+		}
+		return b[:n], nil
+	}
+}
+func (m *SamplingStrategyParameters) XXX_Merge(src proto.Message) {
+	xxx_messageInfo_SamplingStrategyParameters.Merge(m, src)
+}
+func (m *SamplingStrategyParameters) XXX_Size() int {
+	return m.Size()
+}
+func (m *SamplingStrategyParameters) XXX_DiscardUnknown() {
+	xxx_messageInfo_SamplingStrategyParameters.DiscardUnknown(m)
+}
+
+var xxx_messageInfo_SamplingStrategyParameters proto.InternalMessageInfo
+
+func (m *SamplingStrategyParameters) GetServiceName() string {
+	if m != nil {
+		return m.ServiceName
+	}
+	return ""
+}
+
+func init() {
```

Comment:
Note that usage of this sampler implementation triggers these `init`-based proto registration calls, but the golang package this lives in is `internal` so these message and enum types cannot be referenced _from_ this generated source, which is going to be a source of conflicts "duplicate enum registered" without a workaround.